lib/preflight/preflighttest.ts

```diff
@@ -110,6 +115,7 @@ export class PreflightTest extends EventEmitter {
   private _connectTiming = new Timer();
   private _sentBytesMovingAverage = new MovingAverageDelta();
   private _packetLossMovingAverage = new MovingAverageDelta();
+  private _progressEvents: string[] = [];
```

Comment:
How about changing _progressEvents to be `Map<time, event>` and also reporting it to insights.